packages/validator/src/services/block.ts

```diff
@@ -225,10 +232,18 @@ export class BlockProposingService {
         this.logger.error("Failed to produce execution block", {}, fullBlock);
         fullBlock = null;
       }
-    } else {
-      fullBlock = await fullBlockPromise;
+    } else if (blindedBlockPromise !== null && fullBlockPromise === null) {
+      blindedBlock = await blindedBlockPromise;
+      fullBlock = null;
+    } else if (blindedBlockPromise === null && fullBlockPromise !== null) {
       blindedBlock = null;
+      fullBlock = await fullBlockPromise;
+    } else {
+      throw Error(
+        `Neither builder nor execution proposal flow activated: isBuilderEnabled=${isBuilderEnabled} builderSelection=${builderSelection}`
+      );
     }
+    this.logger.info("Started block production", {});
```

Comment:
Let's also add slot to the log, it ensures it shows up if you query for slot specific logs

```suggestion
    this.logger.info("Started block production", {slot});
```

Reply:
oops sorry this log is not necessary

Reply:
ah true, this log would also be incorrect as it is logged after all promises are already awaited